.github/workflows/test-e2e-samples.yml

```diff
@@ -41,8 +41,7 @@ jobs:
         run: |
           KUSTOMIZATION_FILE_PATH="testdata/project-v4/config/default/kustomization.yaml"
           sed -i '25s/^#//' $KUSTOMIZATION_FILE_PATH

```

Comment:
@fischor

To simplify things, I think we should first focus on moving forward with [PR #4123](https://github.com/kubernetes-sigs/kubebuilder/pull/4123).

Key points on top of the changes in this PR:

- In test/testdata/generate.sh, we added commands to ensure comprehensive testing scenarios for single and multi-group layouts. (Check the diffs between the two. Note that the kind of webhooks scaffolds for core and external are not the same in all cases)
- Documentation has been updated accordingly.
- We are skipping scaffolds done for webhooks in config/crd, but I believe we’re missing CA injection for cert-manager. We may need to add validations for external and core types with CA cert-manager injection in the e2e tests.

Feel free to help by:

- Reviewing the changes in PR [#4123](https://github.com/kubernetes-sigs/kubebuilder/pull/4123) to see if we can simplify and move it forward.
- Continuing the work needed to add support for CA injection for cert-manager.

Thanks! Your help is more than welcome 🥇